subworkflows/local/deepvariant_error_polishing/main.nf

```diff
@@ -0,0 +1,312 @@
+include { BCFTOOLS_MERGE                          } from "$projectDir/modules/nf-core/bcftools/merge/main"
+include { BCFTOOLS_CONSENSUS                      } from "$projectDir/modules/nf-core/bcftools/consensus/main"
+include { MERQURY as MERQURY_INPUT_ASM            } from "$projectDir/modules/nf-core/merqury/main"
+include { MERQURY as MERQURY_POLISHED_ASM         } from "$projectDir/modules/nf-core/merqury/main"
+include { DVPOLISH_CREATE_FINALASM                } from "$projectDir/modules/local/dvpolish/createFinalAsm"
+
+/*
+outline: 
+
+|- create pbmm2 index for assembly (1)
+|- create bed chunks for given assembly (1..n)
+|- align all read files to full assembly (1..m)
+    \- split each alignment file to contig according to bed chunks files (n*m)
+    |- in case of multiple read files (therefore optional) merge all read files that belong to same assembly chunk (1..n)
+    |- index merged alignment files (1..n)
+    |- call variants with DeepVartiant (1..n)
+    |- filter variants (PASS + homozygous) (1..n)
+    |- create tabix index files  (1..n)
+    |- merge all variants (1)
+    |- create consensus sequence (1)
+*/
+
+workflow DVPOLISH {
+
+    take:
+    ch_assemblies // [ meta, assembly ]
+    ch_hifi       // [ meta, hifi ]
+    ch_meryl_hifi // [ meta, union.meryldb ] 
+
+    main:
+    
+    reads_plus_assembly_ch = combineByMetaKeys (
+            ch_hifi,
+            ch_assemblies,
+            keySet: ['id','sample'],
+            meta: 'rhs'
+        )
+
+    reads_plus_assembly_ch
+        // Add single_end for minimap module
+        .flatMap { meta, reads, assembly -> reads instanceof List ?
+            reads.collect{ [ meta + [ single_end: true ], it, assembly.pri_fasta ] }
+            : [ [ meta + [ single_end: true ], reads, assembly.pri_fasta ] ] }
+        .multiMap { meta, reads, assembly ->
+            reads_ch: [ meta + [ readID: reads.baseName ], reads ]
+            assembly_ch: [ meta, assembly ]
+        }
+        .set { input }
+
+    uniq_assembly_ch = getPrimaryAssembly(ch_assemblies)
+
+    assembly_plus_meryl_ch = combineByMetaKeys (
+            ch_meryl_hifi,
+            uniq_assembly_ch,
+            keySet: ['id','sample'],
+            meta: 'rhs'
+        )
+
+    // index assembly file(s)
+    SAMTOOLS_FAIDX (
+        uniq_assembly_ch,
+        [[],[]]
+    )    
+
+    // split assembly into smaller chunks, this step just creates bed files 
+    // that represent the assembly chunks, no sequence is split
+    DVPOLISH_CHUNKFA (
+        SAMTOOLS_FAIDX.out.fai
+    )
+
+    // create minimap2 index for assemblies
+    DVPOLISH_PBMM2_INDEX (
+        uniq_assembly_ch
+    )
+
+    // map reads with pbmm2 to complete assemblies (chunks are not used in that step)
+    DVPOLISH_PBMM2_ALIGN (
+        input.reads_ch,
+        input.assembly_ch
+    )
+
+    def path_closure = {meta, files -> files.collect(){[meta, it ]}}
```

Comment:
If I'm not missing something, `transpose` operator does the same thing as `flatMap(path_closure)`. Not sure why I missed this before.